### PR TITLE
Authenticator performance

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Pull changes from https://github.com/ThePalaceProject/circulation-core/pull/44 into circulation.

## Motivation and Context

- Makes authenticator loading faster by warming the cache with the needed integration settings.
- Makes `setup_adobe_vendor_id` use less costly by short-circuiting a loop as soon as conditions are satisfied.

### CODE REVIEW

File: api/routes.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from functools import update_wrapper, wraps
 
 import flask
-from flask import Response, make_response, redirect, request
+from flask import Response, g, make_response, redirect, request
```
Comment (tdilauro) [inline api/routes.py:6]: `g` doesn't appear to be used anymore.

`redirect`, which was already present, also appears not to be used.

**Comment by tdilauro** [comment]
I added a commit for reducing the `setup_adobe_vendor_id` run from `load_settings`. The elapsed time from the `pyinstrument`ed data was 4.2 seconds and in my own local baselines it hovered around 4 seconds. I was able to get this down to on the order of 0.75s in my environment by splitting up the per-library configuration loops and short circuiting the check that used `setup_adobe_vendor_id`. There are still improvements that can and should be made under the hood here, but this seemed like a good start. Attempts at pre-caching the values did not yield any consistent benefit and did not justify the extra complexity.

I also added a commit to pick up an elapsed time logging context manager that was added to `core`.

**Review by tdilauro — APPROVED** [review]
LGTM! 🚀 

A couple minor things to keep the code clean, but nothing that would prevent merging.

File: api/routes.py [hunk 2]
```diff
@@ -48,6 +48,15 @@ def get_locale():
     return request.accept_languages.best_match(languages)
 
 
+@app.after_request
+def print_cache(response):
+    if hasattr(app, "_db") and HasFullTableCache.CACHE_ATTRIBUTE in app._db.info:
```
Comment (vbessonov) [inline api/routes.py:53]: Can it add additional overhead? Can we add an additional check to run it only if `DEBUG` level was turned on?

**Review by vbessonov — APPROVED** [review]

**Pull request merged**